From 50386f8c0630a7ed79072876d69b069720f0420c Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Wed, 6 Nov 2024 10:26:55 +0000
Subject: [PATCH] dt-bindings: usb: snps,dwc3: add FS/HS periodic NAK polling
 quirk

Add two quirk properties that control whether or not the controller
issues many more handshakes to FS/HS Async endpoints in a single
(micro)frame. Enabling these can significantly increase throughput for
endpoints that frequently respond with NAKs.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 Documentation/devicetree/bindings/usb/snps,dwc3.yaml | 10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/Documentation/devicetree/bindings/usb/snps,dwc3.yaml
+++ b/Documentation/devicetree/bindings/usb/snps,dwc3.yaml
@@ -232,6 +232,16 @@ properties:
     description: When set, disable u2mac linestate check during HS transmit
     type: boolean
 
+  snps,enhanced-nak-fs-quirk:
+    description:
+      When set, the controller schedules many more handshakes to Async FS
+      endpoints, improving throughput when they frequently respond with NAKs.
+
+  snps,enhanced-nak-hs-quirk:
+    description:
+      When set, the controller schedules many more handshakes to Async HS
+      endpoints, improving throughput when they frequently respond with NAKs.
+
   snps,parkmode-disable-ss-quirk:
     description:
       When set, disable park mode for all Superspeed bus instances.
